### ISSUE TITLE
Nodejs API doesn't work?

### ISSUE THREAD

**Issue by josephrocca** [luminati-io/luminati-proxy#42]
The docs say to do this:
```javascript
const Luminati = require('luminati-proxy').Luminati;
```
But `require('luminati-proxy')` has no property called `Luminati`. I tried just doing `const Luminati = require('luminati-proxy')` but then I get an error `proxy doesn't have a method called 'on'`.

Also, does this lib use `request-promise`? If not, it should, because then we can replace this:
```javascript
proxy.listen(0, '127.0.0.1').then(()=>new Promise((resolve, reject)=>{
    proxy.request('http://lumtest.com/myip', (err, res)=>{
        if (err)
            return reject(err);
        resolve(res);
    });
})).then(res=>{
    console.log('Result:', res.statusCode, res.body);
}, err=>{
    console.log('Error:', err);
}).then(()=>proxy.stop());
```
With something like this:
```javascript
await proxy.listen(0, '127.0.0.1')
let res = await proxy.request('http://lumtest.com/myip').catch(/* ... */)
console.log('Result:', res.statusCode, res.body);
proxy.stop()
```
And we can all try to forget we ever saw that original code snippet.

**Comment by eyalcohen4** [comment]
Same problems here, tell me that Luminati is not a constructor.
any solution?

**Comment by idietmoran** [comment]
make sure you `npm install luminati-io/luminati-proxy` instead of just `npm install luminati-proxy`

two different packages, and the second one has the same error you have.

**Comment by ovidiuD** [comment]
node 7 is not yet officially supported, so async/await isn't either
for the moment you can use generator syntax as in the README example

`'use strict';
const etask = require('hutil').etask;
const Luminati = require('luminati-proxy').Luminati;

etask(function*(){
    const proxy = new Luminati({
        customer: 'CUSTOMER', // your customer name
        password: 'PASSWORD', // your password
        zone: 'gen', // zone to use
        proxy_count: 5, //minimum number of proxies to use for distributing requests
    });
    yield proxy.listen(0, '127.0.0.1'); // port and ip to listen to
    let res = yield etask.nfn_apply(proxy, '.request',
        ['http://lumtest.com/myip']);
    console.log('Result:', res.statusCode, res.body);
    yield proxy.stop();
});`

**Comment by josephrocca** [comment]
@ovidiuD Your library already uses promises and that's all that's required for async/await (it's just sugar), so it'd be cool if you used [request-promise](https://www.npmjs.com/package/request-promise) for `proxy.request` so that your lib was consistent and easier to use. `request-promise` is just a wrapper around `request` and is well maintained and used (greater than 1 million downloads per month). This is probably best posted as a separate issue though.

**Comment by idietmoran** [comment]
@josephrocca 

http://bluebirdjs.com/docs/api/promise.promisify.html

http://bluebirdjs.com/docs/api/promise.promisifyall.html

or just fork it yourself and make a pull request

**Comment by josephrocca** [comment]
I'm not actually using luminati anymore, but that would be a good workaround in the mean time :+1:

**Comment by hamzaahmad** [comment]
can you help guys how to use array of url ?

**Comment by ovidiuD** [comment]
Hi, can you be more specific into what are you trying achieve, what do you mean by array of url?

**Comment by hamzaahmad** [comment]
i have array contain 25000 url i wanna read with luminati-proxy and get the response , i try to use callback function using each but i got error heap out of memory ?any way to pass  array  for luminati-proxy?

or can we use it with nightmarejs ?
 here is my code with nightmare js

const express = require("express"); 
const Nightmare = require("nightmare");
const cheerio = require("cheerio");
const request = require("request");
const fs = require("fs");

var proxyNightmare = Nightmare({
  switches: {
    'proxy-server': 'my_proxy_server.example.com:8080' // set the proxy server here ...
  },
  show: true
});


const app = express();
var results = [];



var array = fs.readFileSync("LINKS/part1.txt").toString().split(","); // links folder
for (i in array) {
  results.push(array[i]);
}
var array2 = fs.readFileSync("useragent.txt").toString().split("\n"); // links folder
for (i in array2) {
  console.log(array2[i])
  useragent.push(array2[i]);
}
console.log("You have  add " + array.length + " website");
results
  .reduce(function(accumulator, url) {
          var randomValue = useragent[Math.floor(useragent.length * Math.random())];
    return accumulator.then(function(results) {

      return nightmare
        .goto(url)
        .useragent(randomValue)
        .wait(Math.floor(Math.random() * 50000 + 1200)) // random wait to avoid request pattren
        .evaluate(function() {
          return document.body.innerHTML;
        })
        .catch(function(err) {
          console.log("You have Error with", err);
        })
        .then(function(body) {
          var $ = cheerio.load(body);
          $("dd").each(function(i, element) {
            console.log($(element).text());
          });
          $("table").each(function(i, element) {
            console.log($(element).text());
          });
        })
        .catch(function(err) {
          console.log("You have Error with", err);
        });
    });
  }, Promise.resolve([]))
  .then(function(results) {});

const port = process.env.PORT || 3000;

app.listen(port, function() {
  console.log("Server is listen on Port..." + port);
});

**Comment by ovidiuD** [comment]
No there's no such url passing option. It's just a regular proxy.
Your code however seems to be the problem, more exactly the reduce call on results array.
If results contains 25000 items, it means you'll be created 25000 promises waiting on each other, so it could be a problem memory wise.
get rid of the reduce call, and start on with the first request, in then callback do the next one, ...

Ex:
```javascript
(function get_url(i=0, prev_res=null){
new Promise((resolve, reject)=>{
...do_work... with results[i] then resolve/reject async
resolve(some_res);
}).then(res=>get_url(i+1, res));
})();
```

**Comment by ovidiuD** [comment]
closing due to no activity for more than 2 weeks
